packages/api/src/providers/fitbit.ts

```diff
@@ -363,74 +380,57 @@ export class Fitbit extends Provider implements OAuth2 {
   }
 
   async postAuth(token: string, user: ConnectedUser): Promise<void> {
-    const { access_token: accessToken, user_id: fitbitUserId, scope: scopes } = JSON.parse(token);
-
-    const subscriptionTypes: Record<FitbitScopes, FitbitCollectionTypesWithoutUserRevokedAccess> = {
-      [FitbitScopes.activity]: "activities",
-      [FitbitScopes.nutrition]: "foods",
-      [FitbitScopes.sleep]: "sleep",
-      [FitbitScopes.weight]: "body",
-    };
-
-    const activeSubscriptions = await this.getActiveSubscriptions(accessToken);
-    if (activeSubscriptions)
-      await this.revokeTokenFromOtherUsers(user, fitbitUserId, activeSubscriptions, accessToken);
-
-    // TODO #652: Implement userRevokedAccess
+    const fitbitToken = parseFitbitToken(token);
+    await this.revokeTokenFromOtherUsers(user, fitbitToken.userId);
+    await this.createSubscriptions(fitbitToken);
+  }
 
-    // Creates new WH subscriptions based on the user's selected scopes
-    await Promise.all(
-      Object.entries(subscriptionTypes).map(async ([key, subscriptionType]) => {
-        if (scopes.includes(key)) {
-          const subscriptionId = `${fitbitUserId}-${subscriptionType}`;
-          const subscriptionUrl = `${Fitbit.URL}/${Fitbit.API_PATH}/${subscriptionType}/apiSubscriptions/${subscriptionId}.json`;
-          this.createSubscription(subscriptionUrl, accessToken);
+  /**
+   * Finds all users connected to the same Fitbit user, removes their tokens and deletes their WH subscriptions, if there are any.
+   *
+   * @param currentUser current ConnectedUser who is connecting to Fitbit
+   * @param fitbitUserId Fitbit user ID, it might be the same for different ConnectedUsers
+   */
+  async revokeTokenFromOtherUsers(currentUser: ConnectedUser, fitbitUserId: string): Promise<void> {
+    const connectedUsers = await getConnectedUsersByTokenOrFail(
+      ProviderSource.fitbit,
+      fitbitUserId
+    );
+    const rejected: { userId: string; cxId: string; err: unknown }[] = [];
+    await Promise.allSettled(
+      connectedUsers.map(async user => {
+        if (user.dataValues.id !== currentUser.dataValues.id) {
+          try {
+            const token = getProviderTokenFromConnectedUserOrFail(user, ProviderSource.fitbit);
+            const fitbitToken = parseFitbitToken(token);
+            await this.oauth.revokeLocal(user);
+            const updatedUser = await getConnectedUserOrFail({
+              id: user.id,
+              cxId: user.cxId,
+            });
```

Comment:
Yes, because we're now reporting their updated providerMap with the `sendProviderDisconnected`, and the way that function is set up, we have to fetch the user before sending it over